File: dotnet/Allmark/Inline/InsertionRule.cs
namespace Allmark.Inline;

using Allmark.Types;

public static class InsertionRule
{
    public static InlineRule Create()
    {
        return new InlineRule
        {
            Name = "insertion",
            Test = TestInsertion,
            Precedence = 20,
        };
    }

    private static bool TestInsertion(InlineParserState state, MarkdownNode parent)
    {
        var rule = Create();
        return CriticMarksRule.Execute("insertion", "+", state, parent, rule.Precedence!.Value);
    }
}

File: zig/src/inline/insertionRule.zig
const std = @import("std");
const InlineParserState = @import("../types/InlineParserState.zig").InlineParserState;
const InlineRule = @import("../types/InlineRule.zig").InlineRule;
const MarkdownNode = @import("../types/MarkdownNode.zig").MarkdownNode;
const testCriticMarks = @import("criticMarksRule.zig").testCriticMarks;

pub fn testInsertion(state: *InlineParserState, parent: *MarkdownNode) bool {
    return testCriticMarks("insertion", '+', state, parent);
}

pub const insertionRule = InlineRule{
    .name = "insertion",
    .@"test" = testInsertion,
};
